src/Specification/Filter/Equals.php

```diff
@@ -1,20 +1,37 @@
 <?php
 
-/**
- * Spiral Framework. PHP Data Grid
- *
- * @license MIT
- * @author  Anton Tsitou (Wolfy-J)
- * @author  Valentin Vintsukevich (vvval)
- */
-
 declare(strict_types=1);
 
 namespace Spiral\DataGrid\Specification\Filter;
 
 /**
- * Defines that field must match the given value.
+ * Filters records where a field exactly matches the given value.
+ *
+ * ```
+ * // Find users with specific role
+ * $roleFilter = new Equals('role', 'admin');
+ * ```
+ * ```
+ * // Dynamic category filtering
+ * $categoryFilter = new Equals('category', new StringValue());
+ * $result = $categoryFilter->withValue('electronics');
+ * ```
+ * ```
+ * // Order status filtering
+ * $statusFilter = new Equals('status', new EnumValue(
+ *     new StringValue(), 'pending', 'processing', 'shipped', 'delivered'
+ * ));
+ * $result = $statusFilter->withValue('shipped');
+ * ```
+ * ```
+ * // Boolean flag filtering
+ * $activeFilter = new Equals('is_active', new BoolValue());
+ * $result = $activeFilter->withValue(true);
+ * ```
+ * ```
+ * // Numeric ID filtering
+ * $userFilter = new Equals('user_id', new IntValue());
+ * $result = $userFilter->withValue(123);
+ * ```
```

Comment:
```suggestion
 * ```
 * // Find users with specific role
 * $roleFilter = new Equals('role', 'admin');
 *
 * // Dynamic category filtering
 * $categoryFilter = new Equals('category', new StringValue());
 * $result = $categoryFilter->withValue('electronics');
 *
 * // Order status filtering
 * $statusFilter = new Equals('status', new EnumValue(
 *     new StringValue(), 'pending', 'processing', 'shipped', 'delivered'
 * ));
 * $result = $statusFilter->withValue('shipped');
 *
 * // Boolean flag filtering
 * $activeFilter = new Equals('is_active', new BoolValue());
 * $result = $activeFilter->withValue(true);
 *
 * // Numeric ID filtering
 * $userFilter = new Equals('user_id', new IntValue());
 * $result = $userFilter->withValue(123);
 * ```
```

In such cases it might be better to merge samples into one block